### DISCUSSION TITLE
Automation Control not showing Record Triggered Flows

### DISCUSSION THREAD

**Discussion by GADXNghia** [jetstreamapp/jetstream#1029] [Q&A] [answered]
New user here, I am a Junior Salesforce Admin
I just started to try out Jetstream features yesterday.
When I go to Automation Control, I found all the automation display correctly except for the Record Triggered Flows which are empty across my Objects.
On the document, I did not find anything extra steps or further setting I need to do. 
What went wrong?

**Comment by paustint** [comment]
@GADXNghia - hmm, there has been one other user reporting the same behavior, but for some reason we have been unable to reproduce.

1. What edition of Salesforce are you using?
2. Are the flows in question part of a managed package?

Can you run this query in the query builder and see if records are returned?
(Use the "Manual Query" option and paste this in)

This one should return all record-triggered flow across all objects
```sql
SELECT Id, ManageableState, IsTemplate, ActiveVersionId, Label, ApiName, Description,
	DurableId, IsActive, LastModifiedBy, FORMAT(LastModifiedDate), LatestVersionId,
	ProcessType, TriggerObjectOrEventId, TriggerObjectOrEvent.QualifiedApiName,
	TriggerObjectOrEventLabel, TriggerType,
	(
		SELECT Id, ApiVersion, ApiVersionRuntime, DurableId, FORMAT(LastModifiedDate),
			ProcessType, RunInMode, Description, Label, VersionNumber, Status
		FROM Versions
	)
FROM FlowDefinitionView
WHERE ManageableState IN ('unmanaged', 'installed')
	AND TriggerType LIKE 'Record%'
ORDER BY TriggerObjectOrEventId, Label
```


The actual query that Jetstream runs for automation control is like this where there is a list of objects included (in this example `Account` and `Contact`)

```sql
SELECT Id, ManageableState, IsTemplate, ActiveVersionId, Label,
	ApiName, Description, DurableId, IsActive, LastModifiedBy,
	FORMAT(LastModifiedDate), LatestVersionId, ProcessType, TriggerObjectOrEventId,
	TriggerObjectOrEvent.QualifiedApiName, TriggerObjectOrEventLabel, TriggerType,
	(
		SELECT Id, ApiVersion, ApiVersionRuntime, DurableId,
			FORMAT(LastModifiedDate), ProcessType, RunInMode, Description, Label, VersionNumber,
			Status
		FROM Versions
	)
FROM FlowDefinitionView
WHERE TriggerObjectOrEvent.QualifiedApiName IN ('Account', 'Contact')
	AND ManageableState IN ('unmanaged', 'installed')
	AND TriggerType LIKE 'Record%'
ORDER BY TriggerObjectOrEventId, Label
```

**Reply by GADXNghia** [reply]
@paustint 
Hi, 
Good to know that there is someone else have the same issue with me.

1. We are running "Enterprise Edition" 
2. All the record trigger flows is missing from the view, not just a particular one.
3. I can run the first querry
`SELECT Id, ManageableState, IsTemplate, ActiveVersionId, Label, ApiName, Description,
	DurableId, IsActive, LastModifiedBy, FORMAT(LastModifiedDate), LatestVersionId,
	ProcessType, TriggerObjectOrEventId, TriggerObjectOrEvent.QualifiedApiName,
	TriggerObjectOrEventLabel, TriggerType,
	(
		SELECT Id, ApiVersion, ApiVersionRuntime, DurableId, FORMAT(LastModifiedDate),
			ProcessType, RunInMode, Description, Label, VersionNumber, Status
		FROM Versions
	)
FROM FlowDefinitionView
WHERE ManageableState IN ('unmanaged', 'installed')
	AND TriggerType LIKE 'Record%'
ORDER BY TriggerObjectOrEventId, Label`
as it return all the flows

**Reply by paustint** [reply, marked as answer]
@GADXNghia - Any chance you could create an empty sandbox with the flows and invite support@getjetstream.app as a user?
That would help troubleshoot.

A couple other debug options:

**Any output in console?**
* Enable logging (go to your settings in Jetstream and you will see a logging option)
* Then go run the process again
* In the browser console (see screenshot) you should see some log entries that log the request and response of the queries above along with some other stuff
* Are there any errors? when you expand the request responses do you see your flows there? (maybe need to expand a few times and explore a little bit)

![image](https://github.com/user-attachments/assets/7ab0233d-6e07-4c96-a57f-dcbac722da06)


**Record a jam**
Use https://jam.dev/ to do a screen share and show the behavior how nothing is working. Send a link to support@getjetstream.app
Enable logging in your console as described above first as this may get included in the jam and can aid troubleshooting


Also - the other user that reported this seemed to have the problem intermittently - in a new incognito window it would work initially, then it would stop working.
Not sure if that is your experience or not.

**Reply by GADXNghia** [reply]
@paustint Some how I am no longer facing the issue. maybe it takes a few days for the metadate fully loaded???
